lucene/queries/src/java/org/apache/lucene/queries/payloads/PayloadMatcherFactory.java

```diff
@@ -0,0 +1,249 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.lucene.queries.payloads;
+
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import org.apache.lucene.queries.payloads.SpanPayloadCheckQuery.MatchOperation;
+import org.apache.lucene.queries.payloads.SpanPayloadCheckQuery.PayloadType;
+import org.apache.lucene.util.ArrayUtil;
+import org.apache.lucene.util.BytesRef;
+
+/**
+ * Creates a payload matcher object based on a payload type and an operation. PayloadTypes of
+ * INT,FLOAT, or STRING are supported. Inequality operations are supported.
+ */
+public class PayloadMatcherFactory {
+
+  private static final HashMap<PayloadType, HashMap<MatchOperation, PayloadMatcher>>
+      payloadCheckerOpTypeMap;
+
+  static {
+    payloadCheckerOpTypeMap = new HashMap<PayloadType, HashMap<MatchOperation, PayloadMatcher>>();
+    // ints
+    HashMap<MatchOperation, PayloadMatcher> intCheckers =
```

Comment:
Can you use `java.util.EnumMap` for these maps?

Reply:
Now that MatchOperation is an enum, that makes a lot of sense.  I just made that update.  I looked into using a lambda, but I think it got messy because I'm actually using inheritance in the PayloadMatcherFactory for the various subclasses to handle the various types.